tests/test_cli.py

```diff
@@ -126,6 +127,37 @@ async def test_list_devices(discovery_mock, runner):
     assert row in res.output
 
 
+async def test_discover_raw(discovery_mock, runner, mocker):
+    """Test that device update is called on main."""
+    wrapped = redact_data
+    with patch(
+        "kasa.protocols.protocol.redact_data", side_effect=wrapped
+    ) as redact_spy:
```

Comment:
Better use `mocker.patch()` to avoid one level of indent?

Reply:
Done